### PR TITLE
fix: Limit content width DOCS-74

### PR DESCRIPTION
**Pull request by prcr**
Limits the maximum width of the content in the documentation pages. I used the information from [this readability study](https://stackoverflow.com/a/7990214/15377401) to inform the maximum width that the lines of text should have, although I added a bit more space so that the screenshots still have a sufficient size to be clearly readable as well.

The search bar and the footer still have a 100% width, and this can look a bit off. However, I still think that the readability/usability improvement of limiting the content width is worth this simple update. :raised_hands: 

Live preview:
https://deploy-preview-958--docs-codacy.netlify.app/

### CODE REVIEW

**Comment by prcr** [comment]
The site can look a bit weird because the search bar and the footer still have a 100% width. However, I still think that the readability/usability improvement of limiting the content width is worth this update. :raised_hands:

**Comment by inesgomas** [comment]
I understand this limit for a block of text, not for the other content of the page 🤔 

Like the most popular topics being cut like this:
image

What about add this limit to all <p>?

**Comment by prcr** [comment]
> I understand this limit for a block of text, not for other content of the page thinking
> 
> Like the most popular topics being cut like this:

Good catch @inesgomas, I was paying more attention to "full pages" [like this](https://61bcb36c5f1c3f0007858e5d--docs-codacy.netlify.app/getting-started/codacy-quickstart/), and I didn't notice the weird cutoff effect on the homepage "Most popular" cards.

> What about add this limit to all `<p>`?

Here the idea was to limit the context width for all kinds of elements on a page: paragraphs, lists, tables, images, admonition cards, and so on.

So, instead, I should try to figure out why the width of the homepage cards isn't distributed evenly when there's a limit to the width of the containing block. :thinking:

**Comment by prcr** [comment]
@inesgomas, please have a look at the [adjustments I made to the homepage cards](https://github.com/codacy/codacy-mkdocs-material/pull/46/commits/28007277aa27befef655133054081f0a7c78cbd8). Essentially, they're now taking advantage of the Flex capabilities to get themselves distributed evenly across the available page width. :raised_hands: 

You can see how the homepage looks like now:
https://deploy-preview-958--docs-codacy.netlify.app/

File: src/assets/stylesheets/layout/_topics.scss [hunk 1]
```diff
@@ -38,7 +34,6 @@
         font-weight: bold;
         margin-bottom: 0.5rem;
         font-size: 0.8rem;
-        white-space: nowrap;
```
Comment (prcr) [inline src/assets/stylesheets/layout/_topics.scss:41]: This was forcing each card to have at least the width of the title. But it also caused some cards to be wider than the others, which looked worse than having the titles wrapping across two lines.

File: src/assets/stylesheets/layout/_topics.scss [hunk 2]
```diff
@@ -9,18 +9,14 @@
     border: 1px solid #6695FF;
     padding: 1rem;
     border-radius: 0.25rem;
-    flex-basis: 33%;
-    margin: 0 1rem;
```
Comment (prcr) [inline src/assets/stylesheets/layout/_topics.scss:13]: The property `flex-basis` set to a value smaller than 33%, together with `justify-content: space-between;` ensures that there's a small space between the cards. And the `margin` isn't needed anymore. :raised_hands:

File: src/assets/stylesheets/layout/_topics.scss [hunk 3]
```diff
@@ -9,18 +9,14 @@
...
+    flex-basis: 32%;
     text-decoration: none;
     color: var(--codacy-primary);
     &:hover, &:focus {
       color: var(--codacy-primary);
       box-shadow: 0px 6px 5px #6695ff26;
     }
 
-    &:first-child { margin-left: 0 }
-    &:last-child { margin-right: 0 }
```
Comment (prcr) [inline src/assets/stylesheets/layout/_topics.scss:22]: The `justify-content: space-between;` ensures that the first and last cards are stuck to the left and right-hand sides of their container, so it's not necessary to specify these margins.

**Review by inesgomas — DISMISSED** [review]
I still don't think the cards should be forced to that size... They should take the full width of the page. Like the footer does. But I will not block this

**Comment by prcr** [comment]
> I still don't think the cards should be forced to that size... They should take the full width of the page. Like the footer does. But I will not block this

@inesgomas The original issue that I'm fixing in this pull request is that the documentation pages become hard to read if one is using a very large display or browser window. Since we weren't limiting the maximum width of the content, the lines of text can become very long, images become huge, and so on. [DOCS-74](https://codacy.atlassian.net/browse/DOCS-74).

As a side effect of the quick fix :sweat_smile: I'm implementing, I agree that letting the footer as is makes the pages look weird (although only when viewed on very wide viewports). In particular, the homepage is where this effect is probably the most visible. :thinking: 

So, as a further improvement, I had the idea of also limiting the width for the content in the footer in the same way as in the main content of the pages. Like this, in wider viewports, all the content on the page (body and footer) is kept nice and tidy with the same maximum width. I think it looks pretty good now:

image

What do you think?

**Comment by inesgomas** [comment]
At least now they seem to follow the same rule, looks good

**Review by inesgomas — APPROVED** [review]

**Pull request merged**